src/app/space/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.ts

```diff
@@ -123,39 +122,44 @@ export class CodebasesItemWorkspacesComponent implements OnDestroy, OnInit {
   }
 
   /**
-   * Open workspace in editor
+   * Opens Eclipse Che workspace in a new tab
    */
   openWorkspace(): void {
-    this.workspaceBusy = true;
-    this.subscriptions.push(this.cheService.getState().switchMap(che => {
-      if (!che.clusterFull) {
-        // create
-        return this.workspacesService.openWorkspace(this.workspaceUrl).map(workspaceLinks => {
-          this.workspaceBusy = false;
-          let workspaceWindow = this.windowService.open('about:blank', '_blank');
-          if (workspaceLinks != undefined) {
-            workspaceWindow.location.href = workspaceLinks.links.open;
-          }
-        });
-      } else {
-        this.workspaceBusy = false;
-        // display error message
-        this.notifications.message({
-          message: `OpenShift Online cluster is currently out of capacity, workspace cannot be started.`,
-          type: NotificationType.DANGER
-        } as Notification);
-        return Observable.of({});
-      }
-    }).subscribe(() => {},
-      err => {
-        this.workspaceBusy = false;
-        this.handleError('Failed to open workspace', NotificationType.DANGER);
-      }));
+    if (this.eclipseCheUrl) {
+      this.windowService.open(this.eclipseCheUrl, '_blank');
+    } else {
+      this.handleError('Failed to open workspace', NotificationType.DANGER);
+    }
   }
 
+  /**
+   * When combobox element is selected, sets the selected workspace and Eclipse Che url
+   */
   setWorkspaceSelected(): void {
     this.workspaceSelected = (this.workspaceUrl !== undefined && this.workspaceUrl.length !== 0
       && this.workspaceUrl !== 'default');
+    this.setEclipseCheUrl();
+  }
+
+  /**
+   * Set the URL to the Eclipse Che workspace
+   */
+  setEclipseCheUrl(): void  {
+    if (this.workspaceUrl !== 'default') {
+      this.workspaceBusy = true;
+      this.subscriptions.push(this.cheService.getState().switchMap(che => {
+        return this.workspacesService.openWorkspace(this.workspaceUrl).map(workspaceLinks => {
```

Comment:
@aptmac where do you check whether `che.clusterFull` ?

Reply:
`che.clusterFull` is still checked when the workspace is created [0]; this `setEclipseCheUrl` is fired when the user selects an existing workspace in the combobox so the Url can be preloaded and not have to be resolved as an Observable when launching the workspace in a new tab.

[0] https://github.com/aptmac/fabric8-ui/blob/02e6c8447d0911b85d6ce06f9229ac426b18f04e/src/app/space/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.ts#L76

Reply:
@corinnekrych  `clusterFull` should be checked in both places - creating new workspace & opening existing

Reply:
@aptmac let's add it back when opening then....

Reply:
Sure, I'll work on adding that check back, thanks for clarifying.